sql/field.cc

```diff
@@ -4626,17 +4626,17 @@ void Field_longlong::set_max()
   int8store(ptr, unsigned_flag ? ULONGLONG_MAX : LONGLONG_MAX);
 }
 
-bool Field_longlong::is_max()
+bool Field_longlong::is_max(const uchar *ptr_arg) const
```

Comment:
I think the opposite: a pointer should ALWAYS be supplied. And it makes no sense to store ptr in Field (offset should be stored, rather). You already may have noticed, how hard it can be to make literally anything with a record that is not `table->record[0]`. Many things get broken because of that.

In my opinion, these field functions are better not to have any default parameters (or a parameter-less shortcut) so that the one who writes a call would think what are they going to feed there. 

As an example, look at how the code has been simplified in log_event_server.cc once I've forced a correct parameter to pass: turns out, it was just is_max called on data from record[0] and record[1]!

Reply:
Simetimes it simplifies, sometimes not. Most the code got dirtier, I don't like how it looks. The correct paradigm is: fields should be the properies of record, not of the table. Until that I'd prefer record[0] is the default parameter of table fields. Imagine your paradigm in implementation: you supply the offset to *every* property, that are hundreds of them and infinite in infinity. That is horrible!

Reply:
> fields should be the properies of record, not of the table

agree, totally. I had some similar design ideas. I think it'll be easier to transit if most of the code will already accept the correct argument.

Also, can you give your suggestion about how should the changed code in log_event_server.cc look, according to you?

Reply:
> Also, can you give your suggestion about how should the changed code in log_event_server.cc look, according to you?

If that worked you may keep it intact. Or you may apply your solution, I'm not against it as custom measures. I'm against the total conversion of the interface to this paradigm.